theano/gpuarray/opt.py

```diff
@@ -1076,6 +1076,13 @@ def local_gpua_advanced_subtensor(op, context_name, inputs, outputs):
 
 
 @register_opt('fast_compile')
+@op_lifter([tensor.AdvancedBooleanSubtensor])
+@register_opt2([tensor.AdvancedBooleanSubtensor], 'fast_compile')
+def local_gpua_advanced_boolean_subtensor(op, context_name, inputs, outputs):
+    return GpuAdvancedSubtensor()
```

Comment:
All the PR look good, but I have one thing that make me hesitant. On the GPU, you have just one op that also does the boolean indexing. This is different then what we do. I'm not opposed to it, but I would prefer that someone else is also ok about this before merging.

Also, having just one class named GpuAdvancedBooleanSubtensor would be more consistent to what the CPU does. What do you think about this?